nameserver/mdns_client.go

```diff
@@ -272,13 +265,12 @@ func (c *MDNSClient) checkInFlightQueries() time.Duration {
 func (c *MDNSClient) actorLoop(actionChan <-chan MDNSAction) {
 	timer := time.NewTimer(MaxDuration)
 	run := func() { timer.Reset(c.checkInFlightQueries()) }
-	c.running = true
-	for c.running {
+	for {
 		select {
 		case action := <-actionChan:
 			if action == nil {
 				c.listener.Shutdown()
-				c.running = false
+				return
```

Comment:
This changes the logic. By returning instead of breaking the loop, the code below is never executed, and response channels are never closed (probably not important because the client is going to be destroyed and the channels will be closed anyway, but in any case...)